rustls-platform-verifier/src/verification/others.rs

```diff
@@ -54,11 +54,17 @@ impl Verifier {
     /// WebPKI, using root certificates provided by the platform and augmented by
     /// the provided extra root certificates.
     pub fn new_with_extra_roots(
-        roots: impl IntoIterator<Item = pki_types::TrustAnchor<'static>>,
+        roots: impl IntoIterator<Item = impl Into<pki_types::CertificateDer<'static>>>,
     ) -> Self {
         Self {
             inner: OnceCell::new(),
-            extra_roots: roots.into_iter().collect::<Vec<_>>().into(),
+            extra_roots: roots
+                .into_iter()
+                .flat_map(|root| {
+                    webpki::anchor_from_trusted_cert(&root.into()).map(|anchor| anchor.to_owned())
```

Comment:
Is this `to_owned()` call needed? AFAICT `anchor_from_trusted_cert` should be remembering the `'static` lifetime associated with the `CertificateDer<'static>` and returning an "owned" one already.

Reply:
If I try to remove the `to_owned` I get a 
```sh
error[E0515]: cannot return value referencing temporary value
  --> rustls-platform-verifier/src/verification/others.rs:63:34
   |
63 |                 .flat_map(|root| webpki::anchor_from_trusted_cert(&root.into()))
   |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-----------^
   |                                  |                                 |
   |                                  |                                 temporary value created here
   |                                  returns a value referencing data owned by the current function 
```

I guess it's because `webpki::anchor_from_trusted_cert` doesn't take ownership of the certificate, so the even if `CertificateDer` have a `'static` it would be desallocated at the end of the closure so we can't take a reference to it